validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/ProduceBlockRequest.java

```diff
@@ -0,0 +1,235 @@
+      final OkHttpClient okHttpClient,
+      final Spec spec,
+      final UInt64 slot,
+      final boolean preferSszBlockEncoding) {
+    super(baseEndpoint, okHttpClient);
+    this.slot = slot;
+    this.preferSszBlockEncoding = preferSszBlockEncoding;
+    blockContainerSchema = spec.atSlot(slot).getSchemaDefinitions().getBlockContainerSchema();
+    blindedBlockContainerSchema =
+        spec.atSlot(slot).getSchemaDefinitions().getBlindedBlockContainerSchema();
+
+    produceBlockResponseDefinition =
+        DeserializableTypeDefinition.object(ProduceBlockResponse.class)
+            .initializer(ProduceBlockResponse::new)
+            .withField(
+                EXECUTION_PAYLOAD_BLINDED,
+                BOOLEAN_TYPE,
+                ProduceBlockResponse::getExecutionPayloadBlinded,
+                ProduceBlockResponse::setExecutionPayloadBlinded)
+            .withField(
+                EXECUTION_PAYLOAD_VALUE,
+                UINT256_TYPE,
+                ProduceBlockResponse::getExecutionPayloadValue,
+                ProduceBlockResponse::setExecutionPayloadValue)
+            .withField(
+                "data",
+                blockContainerSchema.getJsonTypeDefinition(),
+                ProduceBlockResponse::getData,
+                ProduceBlockResponse::setData)
+            .withField(
+                "version",
+                DeserializableTypeDefinition.enumOf(SpecMilestone.class),
+                ProduceBlockResponse::getSpecMilestone,
+                ProduceBlockResponse::setSpecMilestone)
+            .build();
+
+    produceBlindedBlockResponseDefinition =
+        DeserializableTypeDefinition.object(ProduceBlockResponse.class)
+            .initializer(ProduceBlockResponse::new)
+            .withField(
+                EXECUTION_PAYLOAD_BLINDED,
+                BOOLEAN_TYPE,
+                ProduceBlockResponse::getExecutionPayloadBlinded,
+                ProduceBlockResponse::setExecutionPayloadBlinded)
+            .withField(
+                EXECUTION_PAYLOAD_VALUE,
+                UINT256_TYPE,
+                ProduceBlockResponse::getExecutionPayloadValue,
+                ProduceBlockResponse::setExecutionPayloadValue)
+            .withField(
+                "data",
+                blindedBlockContainerSchema.getJsonTypeDefinition(),
+                ProduceBlockResponse::getData,
+                ProduceBlockResponse::setData)
+            .withField(
+                "version",
+                DeserializableTypeDefinition.enumOf(SpecMilestone.class),
+                ProduceBlockResponse::getSpecMilestone,
+                ProduceBlockResponse::setSpecMilestone)
+            .build();
+
+    produceBlockTypeDefinition =
+        DeserializableOneOfTypeDefinition.object(ProduceBlockResponse.class)
+            .description("meaningful description")
+            .withType(
+                x -> true,
+                s -> s.contains("\"execution_payload_blinded\":false"),
```

Comment:
I have concerns about whitespaces, it is still valid json but fails on this comparison. May be use  `s.matches(".*\"execution_payload_blinded\"\s*:\s*false")`?

Reply:
Done. I guess you meant `s -> s.matches(".*\"execution_payload_blinded\" *: *false.*")` and `s -> s.matches(".*\"execution_payload_blinded\" *: *true.*")`